### PR TITLE
Backport P2P refactor

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
Backports 11363 and 13946

I decided to backport these somewhat out of order so that once v2 P2P system is merged into bitcoin it will be easier to backport.

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
I decided to add more prs related to p2p code and refactoring of it. If you don't think it makes sense to backport some of these prs at this point I can drop them

**Comment by UdjinM6** [comment]
pls see https://github.com/UdjinM6/dash/commits/pr3305

File: src/chainparams.cpp [hunk 1]
```diff
@@ -510,7 +515,7 @@ class CTestNetParams : public CChainParams {
 
         vSeeds.clear();
         // nodes with support for servicebits filtering should be at the top
-        vSeeds.emplace_back("testnet-seed.dashdot.io", true);
+        vSeeds.emplace_back("testnet-seed.dashdot.io"); // Just a static list of stable node(s), only supports x9
```
Comment (codablock) [inline src/chainparams.cpp]: Does anyone know what the `x` and number means? In the BTC PR there are multiple different combinations for seeds.

Comment (UdjinM6) [reply src/chainparams.cpp]: The number is service bits supported by the dns seeder. The idea was that you'd host a seeder on `x<bits>` subdomains to return a list of nodes with only specific bits. Nodes connecting to seeder construct "desired" bits and try to resolve the resulting name. If such subdomain exists - they connect and (presumably) get what they asked for, if not - nodes fallback to connecting to the base domain and get whatever nodes are available.

**Pull request merged**